xtask/src/utils.rs

```diff
@@ -1,5 +1,182 @@
 use serde_json::Value;
-use std::{process::Command, time::Duration};
+use std::{
+    collections::HashMap,
+    path::Path,
+    process::{Child, Command, Stdio},
+    time::Duration,
+};
+
+use crate::{endgroup, group};
+
+// Cargo utils -----------------------------------------------------------
```

Comment:
I would actually refactor the file structure:

```
- src/utils/cargo.rs
- src/utils/workspace.rs
- src/utils/rustup.rs
- src/utils/process.rs
- src/utils/time.rs
```

Reply:
Done !